src/huggingface_hub/commands/repo.py

```diff
@@ -0,0 +1,310 @@
+        )
+        repo_list_parser.add_argument(
+            "--author",
+            type=str,
+            help="Optional: author namespace.",
+        )
+        repo_list_parser.add_argument(
+            "--search",
+            type=str,
+            help="Optional: A string that will be contained in the returned repo ids.",
+        )
+        repo_list_parser.set_defaults(func=lambda args: RepoListCommand(args))
+        repo_delete_parser = repo_subparsers.add_parser("delete", help="Delete a repo on huggingface.co")
+        repo_delete_parser.add_argument(
+            "name",
+            type=str,
+            help="Name of your repo to delete.",
+        )
+        repo_delete_parser.add_argument(
+            "--type",
+            choices=["model", "dataset", "space"],
+            default="model",
+            help="Type of repos to delete, default is model.",
+        )
+        repo_delete_parser.add_argument("--organization", type=str, help="Optional: organization namespace.")
+        repo_delete_parser.add_argument(
+            "-y",
+            "--yes",
+            action="store_true",
+            help="Optional: answer Yes to the prompt",
+        )
+        repo_delete_parser.set_defaults(func=lambda args: RepoDeleteCommand(args))
+        repo_toggle_parser = repo_subparsers.add_parser(
+            "toggle", help="Toggle a repo on huggingface.co private or public"
+        )
+        repo_toggle_parser.add_argument(
+            "name",
+            type=str,
+            help="Name of your repo to toggle.",
+        )
+        repo_toggle_parser.add_argument(
+            "visibility",
+            choices=["public", "private"],
+            default="public",
+            help="Visibility to change repo to, default is public.",
+        )
+        repo_toggle_parser.add_argument(
+            "--type",
+            choices=["model", "dataset", "space"],
+            default="model",
+            help="Type of repo to toggle, default is model.",
+        )
+        repo_toggle_parser.add_argument("--organization", type=str, help="Optional: organization namespace.")
+        repo_toggle_parser.add_argument(
+            "-y",
+            "--yes",
+            action="store_true",
+            help="Optional: answer Yes to the prompt",
+        )
+        repo_toggle_parser.set_defaults(func=lambda args: RepoToggleCommand(args))
+
+
+class BaseRepoCommand:
+    def __init__(self, args: Namespace):
+        self.args = args
+        self._api = HfApi()
+        self.token = HfFolder.get_token()
+        if self.token is None:
+            print("Not logged in")
+            exit(1)
+        try:
+            stdout = subprocess.check_output(["git", "--version"]).decode("utf-8")
+            print(ANSI.gray(stdout.strip()))
+        except FileNotFoundError:
+            print("Looks like you do not have git installed, please install.")
+
+
+class RepoCreateCommand(BaseRepoCommand):
```

Comment:
```suggestion
class RepoCreateCommand(BaseRepoCommand):
    def __init__(self, args: Namespace):
        self.repo_id: str = args.repo_id
        self.repo_type: Optional[str] = args.repo_type
        self.space_sdk: Optional[str] = args.space_sdk
        self.private: bool = args.private
        self.token: Optional[str] = args.token
```

Here is how I would do the `__init__`. By explicitly settings to values here, you still don't validate their type. But in the subsequent code (e.g. when actually calling `create_repo`) you will have an extra insurance that if `__init__` went well then the rest is fine as well (thanks to mypy static check).

In addition to that, I've updated a bit the arguments. I think we should support `--token` for all commands (already the case in `upload`/`download`) as some users might use the CLI in scripts in which they don't want to save a token on the machine (might be a CI machine running in the cloud for example). Also supporting an optional `--private` option (as _store_true_) would be a nice addition.

Finally, I'm not sure we should have this `-y` or `--yes` option. Let's assume that if a user wants to create a repo, then let's create it. I know it is already existing code but I don't expect it to be used a lot.